src/diffuse/vcs/src.py

```diff
@@ -0,0 +1,261 @@
+# Diffuse: a graphical tool for merging and comparing text files.
+#
+# Copyright (C) 2019 Derrick Moser <derrick_moser@yahoo.com>
+# Copyright (C) 2021 Romain Failliot <romain.failliot@foolstep.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import os
+import glob
+
+from gettext import gettext as _
+from typing import Optional, Tuple
+
+from diffuse import utils
+from diffuse.preferences import Preferences
+from diffuse.vcs.folder_set import FolderSet
+from diffuse.vcs.vcs_interface import VcsInterface
+
+
+# SRC support
+class Src(VcsInterface):
+    def __init__(self, root: str):
+        super().__init__(root)
+        self.url: Optional[str] = None
+
+    @staticmethod
+    def _getVcs() -> str:
+        return 'src'
+
+    @staticmethod
+    def _getURLPrefix() -> str:
+        return 'URL: '
+
+    @staticmethod
+    def _parseStatusLine(s: str) -> Tuple[str, str]:
+        # src status prints eg "M<TAB>filename" - as opposed to svn which expands the TAB!
+        if len(s) < 3 or s[0] not in 'ACDMR':
+            return '', ''
+        return s[0], s[2:]
+
+    @staticmethod
+    def _getPreviousRevision(rev: Optional[str]) -> str:
+        if rev is None:
+            return 'BASE'
+        m = int(rev)
+        return str(max(m > 1, 0))
```

Comment:
This appears to have the same error as the SVN code has. I reported it in #227.

Reply:
Hmmm - I had this in my 2018 version (not public, probably derived from the sourceforge code):
+            if m > 1:
+                return str(m - 1)

but I bowed to the svn version when I ported that old cruft to 0.9.0

That said, I think this is only used in conflict resolution which is kinda moot in the way I use SRC (single-user). Probably should be corrected. I wonder why #227 was not fixed.

Reply:
Not conflict resolution, it's for getting the previous revision of the passed argument.
Does this mean that you didn't test all the options?